generate/uuidv7/generator.go

```diff
@@ -0,0 +1,28 @@
+package uuidv7
+
+import googleuuid "github.com/google/uuid"
+
+// UuidV7Generator is the interface for UUID v7 generators.
+type UuidV7Generator interface {
+	Generate() (string, error)
+}
+
+// DefaultUuidV7Generator is the default [UuidGenerator] implementation.
+type DefaultUuidV7Generator struct{}
+
+// NewDefaultUuidV7Generator returns a [DefaultUuidGenerator], implementing [UuidGenerator].
+func NewDefaultUuidV7Generator() *DefaultUuidV7Generator {
+	return &DefaultUuidV7Generator{}
+}
+
+// Generate returns a new UUID V7, using [Google UUID].
+//
+// [Google UUID]: https://github.com/google/uuid
+func (g *DefaultUuidV7Generator) Generate() (string, error) {
+	uuid, err := googleuuid.NewV7()
+	if err != nil {
+		return "", err
+	}
+
+	return uuid.String(), nil
```

Comment:
How aobut returning `uuid.UUID`? Mostly, we're operating with `uuid.UUID` not `string`. :)

Reply:
I can, but what do you do with it except using its string representation ?

Reply:
ah I see, you mean in structs, ok, updating

Reply:
done